nimble/controller/src/ble_ll_scan.c

```diff
@@ -1113,6 +1114,60 @@ ble_ll_scan_window_chk(struct ble_ll_scan_sm *scansm, uint32_t cputime)
     return 0;
 }
 #if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_EXT_ADV)
+static void
+ble_ll_scan_aux_data_free(struct ble_ll_aux_data *aux_scan)
+{
+    if (aux_scan) {
+        if (aux_scan->evt) {
+            ble_hci_trans_buf_free((uint8_t *)aux_scan->evt);
+            aux_scan->evt = NULL;
+        }
+        os_memblock_put(&ext_adv_pool, aux_scan);
+        STATS_INC(ble_ll_stats, aux_freed);
+    }
+}
+
+void
+ble_ll_scan_aux_data_ref(struct ble_ll_aux_data *aux_data)
+{
+    os_sr_t sr;
+
+    OS_ENTER_CRITICAL(sr);
+    if (!aux_data) {
+        OS_EXIT_CRITICAL(sr);
+        return;
+    }
+
+    aux_data->ref_cnt++;
+    STATS_INC(ble_ll_stats, aux_ref);
+    OS_EXIT_CRITICAL(sr);
+}
+
+int
+ble_ll_scan_aux_data_unref(struct ble_ll_aux_data *aux_data)
+{
+    os_sr_t sr;
+    int ret;
+
+    OS_ENTER_CRITICAL(sr);
+    if (!aux_data) {
```

Comment:
probably it would be better to just assert when we try to ref/unref aux_data which is NULL - this is clearly an error

Reply:
well is some part of code this is called  without checking aux_data and this is ok that  aux_data is null then. But maybe I will change that so this function is called only for valid aux_data ... I will see

Reply:
so I think other parts of code should check for null - if you want to ref something, I'd expect it's not null.